suite-native/module-device-settings/src/hooks/useDeviceChangedCheck.ts

```diff
@@ -35,6 +36,7 @@ export const useDeviceChangedCheck = () => {
 
         // Check for device mismatch only after initial device ID is set
         if (initialDeviceIdRef.current && device?.id && initialDeviceIdRef.current !== device.id) {
+            TrezorConnect.cancel();
```

Comment:
I'm not sure this is safe if we don't know what is it canceling.

Reply:
Yeah, you have more knowledge about it, but I assumed, based on its usage in the `ContinueOnTrezor` page, that it’s just canceling any ongoing action on `Trezor`. It’s more like a `cleanup` before leaving the page and as I know it used solely for handling incorrect device that has been connected on Device Setting screen, tbh not sure if there might be other scenariosu where it can caused issues

But if there are any concerns, I’m happy to change it. Thanks!

Reply:
From my testing, it seems to work fine 👌